runway/cfngin/hooks/aws_lambda.py

```diff
@@ -163,6 +173,176 @@ def _zip_from_file_patterns(root, includes, excludes, follow_symlinks):
     return _zip_files(files, root)
 
 
+def _handle_use_pipenv(package_root, dest_path, timeout=300):
+    """Create requirements file from Pipfile.
+
+    Args:
+        package_root (str): Base directory to generate requirements from.
+        dest_path (str): Where to output the requirements file.
+        timeout (int): Seconds to wait for process to complete.
+
+    Raises:
+        PipenvError: Non-zero exit code returned by pipenv process.
+
+    """
+    # TODO stream process output to stdout/stderr
+    LOGGER.info('Creating requirements.txt from Pipfile...')
+    req_path = os.path.join(dest_path, 'requirements.txt')
+    cmd = ['pipenv', 'lock', '--requirements', '--keep-outdated']
+    with open(req_path, 'w') as requirements:
+        pipenv_process = subprocess.Popen(cmd, cwd=package_root,
+                                          stdout=requirements,
+                                          stderr=subprocess.PIPE)
+        _stdout, stderr = pipenv_process.communicate(timeout=timeout)
```

Comment:
I _think_ something like this would work here to allow realtime log streaming but would need to be tested.

```suggestion
        pipenv_process = subprocess.Popen(cmd, cwd=package_root,
                                          stdout=requirements,
                                          stderr=subprocess.PIPE)
        for line in pipenv_process.stderr:
            LOGGER.info(line)
        _stdout, stderr = pipenv_process.communicate(timeout=timeout)
```